app/api/v1/endpoints/exchanges.py

```
@@ -928,3 +928,81 @@ def mask_api_key(api_key: str) -> str:
     if len(api_key) <= 8:
         return "*" * len(api_key)
     return api_key[:4] + "*" * (len(api_key) - 8) + api_key[-4:]
+
+
+async def get_user_portfolio_from_exchanges(user_id: str, db: AsyncSession) -> Dict[str, Any]:
+    """Get user's portfolio data from all connected exchanges using existing balance system."""
+    try:
+        # Get all active exchange accounts for user
+        stmt = select(ExchangeAccount, ExchangeApiKey).join(
```

Ruff (0.12.2)

Undefined name `select`

(F821)